xmppserver/src/main/java/org/jivesoftware/openfire/net/StanzaHandler.java

```diff
@@ -325,6 +325,12 @@ private IQ getIQ(Element doc) {
         Element query = doc.element("query");
         if (query != null && "jabber:iq:roster".equals(query.getNamespaceURI())) {
             return new Roster(doc);
+        }else if (query != null && "jabber:iq:version".equals(query.getNamespaceURI())) {
```

Comment:
There are two issues here:

1. if a result does not include one of the fields, the 'getStringValue()' will cause a NullPointerException
2. If a result includes other fields, we're not registering them.

I suggest you replace this with a loop that iterates over all elements of query.